### PR TITLE
Add cache-control header to uploaded files on S3

### PR DESCRIPTION
**Pull request by Dimitris24sta23**
@tlatsas Hey there!

We need to add a cache-control header in uploaded files in S3
I've added an option and seems to work.

Is it easy for you to review it?

Thanks!

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -21,8 +21,8 @@ def initialize(asset_objects)
       @asset_objects = asset_objects
     end
 
-    def upload!(object_key, file, content_type = "")
-      @asset_objects << { object_key: object_key, file: file, content_type: content_type }
+    def upload!(object_key, file, content_type = "", cache_control)
```
Comment (tlatsas) [inline test/test_helper.rb:24]: I would sugget providing a default value here as well.

**Review by tlatsas — CHANGES_REQUESTED** [review]
Hello @Dimitris24sta23, thanks for the PR! I think it makes sense to add this.

The configuration object `WebpackerUploader::Configuration` is used for holding more global and gem-related configuation options, affecting how this gem works.

Whereas the cache control option is more closely tied to the bucket / uploaded files. Having said that I think it is best to add the cache control option as an option to the `upload!` method.

This is very similar to how the `prefix` is set in the same method. That way we can pass bucket or file-related options there and keep the gem configuration to a bare mininum.

Can you please update the PR to reflect this? You need to remove the option added to the configuration object and add a new parameter named `cache_control` to the `upload!` method in `instance.rb` file. In the readme file add a new section similar to the one [describing the prefix](https://github.com/tlatsas/webpacker_uploader#prefix-remote-files).

File: test/webpacker_uploader_test.rb [hunk 2]
```diff
@@ -29,4 +29,10 @@ def test_upload_with_prefix
     assert_instance_of Pathname, @asset_objects.first[:file]
     assert_equal "text/css", @asset_objects.first[:content_type]
   end
+
+  def test_upload_with_cache_control
+    WebpackerUploader.upload!(@provider, cache_control: "cache_control")
+    puts @asset_objects
```
Comment (tlatsas) [inline test/webpacker_uploader_test.rb:35]: You can remove this `puts` from the test.

**Review by tlatsas — COMMENTED** [review]
Looks good overall! A few more small changes here and there and we can merge this!

I think I can merge and release a new version later today if you send these changes. 🙏

File: lib/webpacker_uploader/version.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module WebpackerUploader
-  VERSION = "0.6.0"
+  VERSION = "0.7.0"
```
[the remark this answers is not in the document]
Comment (Dimitris24sta23) [reply lib/webpacker_uploader/version.rb]: @tlatsas Sorry did this cause i was loading this gem directly from my gitgub to test it.

**Review by tlatsas — APPROVED** [review]
LGTM, thanks for the PR! 🎉

**Pull request merged**